content/utilities/shortest-path/plugin.js

```diff
@@ -0,0 +1,434 @@
+  })
+
+  async function buildGraphClick() {
+    const count = Array.from(df.getAllPlanets()).filter(p => p.planetLevel >= minLevel).length
+    if (window.confirm(`This will create an index of ${count} planets and might take a while. Continue?`)) {
+      buildGraph(minLevel)
+    }
+  }
+
+  return html`
+    <div style="display: flex; justify-content: space-between;">
+        <h1>Shortest Path</h1>
+        <div style="display: flex; gap: 1rem;">
+            <select style=${SelectStyle} value=${minLevel} onChange=${e => setMinLevel(Number(e.target.value))}>
+                <option value="">Min Level</option>
+                ${Object.values(PlanetLevel).map(lvl => html`<option value=${lvl}>${lvl}</option>`)}
+            </select>
+            <button style=${Button} onClick=${buildGraphClick}>
+                Build Graph
+            </button>
+        </div>
+    </div>
+    <p>
+        <i style=${{ ...VerticalSpacing, display: 'block' }}>
+            Visualise the shortest path by distance, time or energy.
+        </i>
+    </p>
+    <${PlotCourse} />
+    <hr style="${VerticalSpacing}" />
+    <div style="display: flex; justify-content: space-between; gap: 1rem;">
+        <button style=${showDistance ? ButtonDistanceOn : ButtonDistance } onClick=${() => setShowDistance(! showDistance)}>
+          Distance
+        </button>
+        <button style=${showTime ? ButtonTimeOn : ButtonTime } onClick=${() => setShowTime(! showTime)}>
+          Time
+        </button>
+        <button style=${showEnergy ? ButtonEnergyOn : ButtonEnergy } onClick=${() => setShowEnergy(! showEnergy)}>
+          Energy
+        </button>
+    </div>
+  `;
+}
+
+class Plugin {
+  constructor() {
+    this.container = null;
+  }
+
+  async render(container) {
+    this.container = container;
+    container.style.width = "480px";
+    render(html`<${App} />`, container);
+  }
+
+  drawPath(viewport, ctx, style, planets) {
+    ctx.beginPath()
+    ctx.strokeStyle = style
+
+    const coords = df.getPlanetsWithIds(planets).map(p => p.location.coords)
+
+    ctx.moveTo(
+        viewport.worldToCanvasX(coords[0].x), 
+        viewport.worldToCanvasY(coords[0].y),
+    )
+
+    for (let coord of coords.slice(1)) {
+        ctx.lineTo(
+            viewport.worldToCanvasX(coord.x), 
+            viewport.worldToCanvasY(coord.y),
+        )
+
+        ctx.moveTo(
+            viewport.worldToCanvasX(coord.x), 
+            viewport.worldToCanvasY(coord.y),
+        )
+    }
+
+    ctx.stroke()
+  }
+
+  draw(ctx) {
+    const viewport = ui.getViewport()
+
+    ctx.lineWidth = 2
+
+    if (ctxShowDistance && paths.distance) this.drawPath(viewport, ctx, distanceStyle, paths.distance)
+    if (ctxShowEnergy && paths.energy) this.drawPath(viewport, ctx, energyStyle, paths.energy)
+    if (ctxShowTime && paths.time) this.drawPath(viewport, ctx, timeStyle, paths.time)
+  }
+
+  destroy() {
+    render(null, this.container);
+  }
+}
+
+/**
+ * And don't forget to it!
```

Comment:
```suggestion
 * And don't forget to export it!
```

or just remove this comment (:

Reply:
@ichub done!